src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -81,18 +82,16 @@ internal class Http1OutputProducer : IHttpOutputProducer, IHttpOutputAborter, ID
             IHttpMinResponseDataRateFeature minResponseDataRateFeature,
             MemoryPool<byte> memoryPool)
         {
-            _pipeWriter = pipeWriter;
+            // Allow appending more data to the PipeWriter when a flush is pending.
+            _pipeWriter = new ConcurrentPipeWriter(pipeWriter, memoryPool);
             _connectionId = connectionId;
             _connectionContext = connectionContext;
             _log = log;
             _minResponseDataRateFeature = minResponseDataRateFeature;
-            _flusher = new TimingPipeFlusher(pipeWriter, timeoutControl, log);
+            _flusher = new TimingPipeFlusher(_pipeWriter, timeoutControl, log);
```

Comment:
I thought we didn't need the TimingPipeFlusher anymore. Unless I totally missed that.